### DISCUSSION TITLE
$watch a parent from child without causing infinite loop?

### DISCUSSION THREAD

**Discussion by yabdab** [alpinejs/alpine#4146] [1. Help]
OK, so I have a parent x-data that has an array of "items".

The child has it's own x-data of values, that  is pushed into this array of "items" inside the parent.

The child will take values from other "items" and recalculate a new value for itself.

So it is dynamic and sync'd I was going to $watch the "items" array and recalculate on-the-fly.

The only problem is this would cause an infinite loop on the "calculation" child inside of "items".

My question is, is there any way to accomplish this without causing the loop to occur?

Hopefully I explained it well enough 😬

**Comment by SimoTod** [comment]
If the child is changing the element that is watching, it will cause an infinite loop, you need to approach the problem in a different way: you probably can do it without watch with a getter. 
If you post a simplify version of your code showing the structure/data you are using, it's probably easier to answer though.

**Comment by ekwoka** [comment]
The problem statement suggests that this is a misuse of $watch.
